### DISCUSSION TITLE
How to remove a panel programmatically ?

### DISCUSSION THREAD

**Discussion by RiadhAdrani** [mathuo/dockview#292] [Q&A] [answered]
I want to remove a panel depending on an external user event, I searched the docs but I didn't find anything, any suggestions ?

**Comment by mathuo** [comment, marked as answer]
There actually isn't a `removePanel` method directly exposed currently so to progmatically remove you would need to do this via the panels `api` methods. For example you can close a panel by calling `.close()`.

```tsx
const panel = api.getPanel(my_panel_id);
panel.api.close();
```

I will add a task to expose a `removePanel`-like method to the primary `api` though as I can see how this would be useful.

Let me know if this helps and if you have any other questions.

**Reply by RiadhAdrani** [reply]
Yep, it worked for me, looking forward to some helper functions.

**Reply by mathuo** [reply]
v1.8.0 now has exposed a `removePanel(panel: IDockviewPanel)` method and is documented [here](https://dockview.dev/docs/next/components/dockview#remove-panel).
